packages/semaphore-group-pcd/src/SemaphoreGroupPCD.ts

```diff
@@ -138,15 +140,23 @@ export async function prove(
     signal: args.signal.value.toString(),
   };
 
-  const proof: SemaphoreGroupPCDProof = {
-    proof: fullProof,
-  };
+  const proof: SemaphoreGroupPCDProof = fullProof.proof;
 
   return new SemaphoreGroupPCD(uuid(), claim, proof);
 }
 
 export async function verify(pcd: SemaphoreGroupPCD): Promise<boolean> {
-  const valid = await verifyProof(pcd.proof.proof, pcd.claim.group.depth);
+  const deserializedGroup = deserializeSemaphoreGroup(pcd.claim.group);
+
+  const fullProof: FullProof = {
+    externalNullifier: pcd.claim.externalNullifier,
+    merkleTreeRoot: deserializedGroup.root + "",
+    nullifierHash: pcd.claim.nullifierHash,
+    signal: pcd.claim.signal,
+    proof: pcd.proof,
+  };
+
+  const valid = await verifyProof(fullProof, pcd.claim.group.depth);
```

Comment:
nice, so `claim.group` members are not needed for verification

one thing i'd recommend here is to either:
- remove `claim.group.members` entirely, or
- have `verify()` reconstruct a group by adding all members and check that the root matches `claim.group.root`

as far as possible, nothing should be the claim that isn't verified in `verify()`

--

we don't necessarily have to do this in this PR though